### PR TITLE
add missing dependencies

### PR DESCRIPTION
**Pull request by timbertson**
(replacement for #5338)

Working on [opam2nix](https://github.com/timbertson/opam2nix), I've encountered a number of dependencies which needed to be added to get certain packages to compile - my assumption is that these dependencies are incidentally provided by transitive dependencies. In `opam2nix` only explicit direct dependencies are visible to the package being compiled, which explains why they would fail under this mode (whereas in normal opam operation, I believe a package can use any package which happens to be installed, even if it doesn't declare a dependency on it).

/cc @gasche @dbuenzli @dsheets @avsm

### CODE REVIEW

**Comment by samoht** [comment]
There's some parsing issue with opam 1.1.2:

```
[WARNING] Errors while parsing biniou.1.0.9 OPAM file, skipping.
[WARNING] install and dev-repo are unknown fields in /home/travis/.opam/repo/default/packages/liquidsoap/liquidsoap.1.1.1/opam: is your OPAM up-to-date ?
[WARNING] Errors while parsing zarith.1.3 OPAM file, skipping.
```

**Comment by samoht** [comment]
Also minios-xen doent' compile on Darwin (/cc @talex5):
```
#=== ERROR while installing minios-xen.0.7 ====================================#
# opam-version 1.2.2
# os           darwin
# command      make debug=n
# path         /Users/travis/.opam/system/build/minios-xen.0.7
# compiler     system (4.02.3)
# exit-code    2
# env-file     /Users/travis/.opam/system/build/minios-xen.0.7/minios-xen-30991-c61e7b.env
# stdout-file  /Users/travis/.opam/system/build/minios-xen.0.7/minios-xen-30991-c61e7b.out
# stderr-file  /Users/travis/.opam/system/build/minios-xen.0.7/minios-xen-30991-c61e7b.err
### stdout ###
# perl include/minios-external/bsd-sys-queue-h-seddery include/minios-external/bsd-sys-queue.h --prefix=minios  >include/list.h.new
# if ! cmp -s include/list.h.new include/list.h; then mv -f include/list.h.new include/list.h; else rm -f include/list.h.new; fi
# ln -sf . include/mini-os
# ln -sf . arch/x86/include/mini-os
# gcc -DCONFIG_START_NETWORK -DCONFIG_SPARSE_BSS -DCONFIG_BLKFRONT -DCONFIG_NETFRONT -DCONFIG_KBDFRONT -DCONFIG_FBFRONT -DCONFIG_CONSFRONT -DCONFIG_XENBUS -nostdinc -isystem include -fno-builtin -Wall -Werror -Wredundant-decls -Wno-format -Wno-redundant-decls -Wformat -fno-stack-protector -fgnu89-inline -Wstrict-prototypes -Wnested-externs -Wpointer-arith -Winline -O3  -D__INSIDE_MINIOS__ -m64 -mno-red-zone -fno-reorder-blocks -fno-stack-protector -fno-asynchronous-unwind-tables -isystem /Users/travis/.opam/system/build/minios-xen.0.7/include -D__MINIOS__ -isystem /Users/travis/.opam/system/build/minios-xen.0.7/include -D__MINIOS__ -D__XEN_INTERFACE_VERSION__=0x00030205  -isystem /Users/travis/.opam/system/build/minios-xen.0.7/arch/x86/include -isystem /Users/travis/.opam/system/build/minios-xen.0.7/arch/x86/include/x86_64 -c blkfront.c -o /Users/travis/.opam/system/build/minios-xen.0.7/blkfront.o
### stderr ###
# clang: error: unknown argument: '-fno-reorder-blocks'
# make: *** [/Users/travis/.opam/system/build/minios-xen.0.7/blkfront.o] Error 1
=-=- Error report -=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=  🐫 
The following actions failed
  ∗  install minios-xen 0.7
The following changes have been performed
  ∗  install conf-perl 1
The former state can be restored with:
    opam switch import "~/.opam/system/backup/state-20160125100727.export"
```

**Comment by samoht** [comment]
Also, https://github.com/ocaml/opam-repository/pull/5766 seems to tackle similar issues but is a more advanced state :-) /cc @OCamlPro-Henry

File: packages/ocurl/ocurl.0.7.6/opam [hunk 1]
```diff
@@ -18,11 +18,12 @@ build-test: [
...
-depends: ["ocamlfind" {build}]
+depends: [
+  "ocamlfind" {build}
+  "conf-libcurl"
+]
 depopts: ["lwt"]
 depexts: [
-  [["debian"] ["libcurl4-gnutls-dev"]]
-  [["ubuntu"] ["libcurl4-gnutls-dev"]]
   [["centos"] ["libcurl-devel" "openssl-devel"]]
```
Comment (gasche) [inline packages/ocurl/ocurl.0.7.6/opam]: I think that you should add the `centos` depexts to `conf-libcurl` to be able to remove all depexts here. The commit introducing the centos depexts is bc430e11f36925540fb7855c092d469eeea4a0a8 , and it quite clearly corresponds exactly to `libcurl4-gnutls-dev` depexts on other systems.

File: packages/zarith/zarith.1.4.1/opam [hunk 2]
```diff
@@ -14,6 +14,7 @@ remove:  ["ocamlfind" "remove" "zarith"]
 depends: [
   "ocamlfind"
   "conf-gmp"
+  "conf-perl" {build}
 ]
 depexts: [
  [["alpine"]["perl"]]
```
Comment (gasche) [inline packages/zarith/zarith.1.4.1/opam]: Same here, I would have expected an `alpine` depext in `conf-perl`, and no depext anymore.

**Comment by timbertson** [comment]
@samoht thanks for the tips. I'm assiming the parse errors mean opam <1.2 doesn't understand the `{build}` flags on deps? Assuming that's the reason, I've taken the flag off these deps. Let me know if I should instead bump the opam version in these files to `1.2` (which seems better for the future, but could potentially break things).

I'm not sure what to do about minios-xen - I'm assuming from that error it already didn't compile on darwin?

@gasche good points - I tried to be conservative because I wasn't sure whether `openssl-devel` really counted as part of libcurl, but I'm more than happy to move all the depexts into conf-* and modify them later if they turn out not to be quite right.

..and thanks @OCamlPro-Henry for your related PR - I'd be more than happy to drop my `ocamlfind` changes from this PR in preference to your PR.

**Comment by samoht** [comment]
Actually I think it's better to just bump the opam version number to `1.2` if possible. Yes you have nothing to do for minios-xen, just maybe add an `available: [ os != "Darwin" ]` if you are at it but that's not mandatory. Thanks!

**Comment by timbertson** [comment]
OK, bumped to 1.2 and added the Darwin exception for minios-xen.

**Comment by OCamlPro-Henry** [comment]
@timbertson I just marged #5276, could you please rebase this one ?

**Comment by timbertson** [comment]
@OCamlPro-Henry done :)

**Comment by OCamlPro-Henry** [comment]
Thanks ! This seems OK, except that the `os` variable in the current OPAM is case-sensitive. It should be  `available: [ os != "darwin" ]` (with a lowercase `d` in `darwin`). Still I will merge this one and fix this in another PR. Thanks again.

**Pull request merged**